### ISSUE TITLE
UX bug: when using count, the plan/apply command and the taint command require different ways to refer to a specific resource instance

### ISSUE THREAD

**Issue by aviflax** [hashicorp/terraform#11045]
I ❤️ Terraform — please consider this a very minor note from an appreciative user.

----

This works:

    $ terraform plan -target=aws_instance.kafka[0]

(as does `apply`)

but this does not:

```
$ terraform taint aws_instance.kafka[0]
The resource aws_instance.kafka[0] couldn't be found in the module root
```

With `taint` I have to use `aws_instance.kafka.0`.

This is annoying.

**Comment by mitchellh** [comment]
Agreeeeeeeed. Thanks.

**Comment by mitchellh** [comment]
Tagged for 0.9 since we'll probably have to introduce backwards incompat here.

**Comment by huydinhle** [comment]
@mitchellh  I would like to work on this. This is my first time trying to contribute to terraform. Thank you

**Comment by mitchellh** [comment]
Hey @huydinhle, that'd be great. I want to give you a few disclaimers up front. First, I think this _might_ be  a tricky issue. It may be easy but also perhaps not. The one thing I know is that we need the `taint` command to use the `ResourceAddress` structure to be uniform with the others. I'm only letting you know it may be tricky as a warning, not pre-judging your ability one way or the other.

Second, if we don't hear from you from some time and this becomes a priority, we may begin working on this ourselves. We'll post to this issue if/when we do. At the moment, there is no plan to do so, but I wanted to put it out there.

If you have any questions at all feel free to ask and I'll be happy to help.

**Comment by huydinhle** [comment]
@mitchellh  Thank you for the heads up. I will try it out and update back to you what I found, whether it's easy or tricky.

**Comment by huydinhle** [comment]
@mitchellh After looking at the code on how terraform did with apply, plan, refresh, I tried to do the same with taint. This is my first try https://github.com/huydinhle/terraform/pull/1/files. This is by no mean a working version. I just post it here to ask to check if I am going to the right direction before I go on and invest more time with this direction.

**Comment by mitchellh** [comment]
I commented on your PR there to try to help! I would not look at apply/plan/refresh for taint since taint is far simpler than those and doesn't require an operation or graph. The details are in that PR though (my comment there).

**Comment by mitchellh** [comment]
I would look instead at `terraform state mv` `terraform state rm` etc which are much more similar to taint in that they 1.) find a thing 2.) make changes to a thing in the state.

**Comment by apparentlymart** [comment]
Hi all!

There is a fix for this in master ready to be included in the forthcoming v0.12.0 release. Sorry for the long silence here!
